src/main/kotlin/app/revanced/patches/all/connectivity/fakewifi/patch/FakeWifiPatch.kt

```diff
@@ -0,0 +1,252 @@
+            "V",
+        ),
+        RequestNetwork2(
+            "Landroid/net/ConnectivityManager;",
+            "requestNetwork",
+            arrayOf("Landroid/net/NetworkRequest;", "Landroid/net/ConnectivityManager\$NetworkCallback;", "I"),
+            "V",
+        ),
+        RequestNetwork3(
+            "Landroid/net/ConnectivityManager;",
+            "requestNetwork",
+            arrayOf("Landroid/net/NetworkRequest;", "Landroid/net/ConnectivityManager\$NetworkCallback;", "Landroid/os/Handler;"),
+            "V",
+        ),
+        RequestNetwork4(
+            "Landroid/net/ConnectivityManager;",
+            "requestNetwork",
+            arrayOf("Landroid/net/NetworkRequest;", "Landroid/app/PendingIntent;"),
+            "V",
+        ),
+        RequestNetwork5(
+            "Landroid/net/ConnectivityManager;",
+            "requestNetwork",
+            arrayOf("Landroid/net/NetworkRequest;", "Landroid/net/ConnectivityManager\$NetworkCallback;", "Landroid/os/Handler;", "I"),
+            "V",
+        );
+
+        fun replaceInstruction(method: MutableMethod, instruction: Instruction35c, instructionIndex: Int) {
+            val registers = arrayOf(instruction.registerC, instruction.registerD, instruction.registerE, instruction.registerF, instruction.registerG)
+            val argsNum = this.methodParams.size + 1 // + 1 for instance of definedClassName
+            if (argsNum > registers.size) {
+                // should never happen, but just to be sure (also for the future) a safety check
+                throw RuntimeException("not enough registers for ${this.definedClassName}#${this.methodName}: required $argsNum registers, but only got ${registers.size}")
+            }
+            val args = registers.take(argsNum).joinToString(separator = ", ") { reg -> "v${reg}" }
+            val replacementMethodDefinition = "${this.methodName}(${this.definedClassName}${this.methodParams.joinToString(separator = "")})${this.returnType}"
+            method.replaceInstruction(
+                instructionIndex,
+                "invoke-static { $args }, ${INTEGRATIONS_CLASS_DESCRIPTOR}->${replacementMethodDefinition}"
+            )
+        }
+
+        companion object {
+            fun fromMethodReference(methodReference: MethodReference) = values().firstOrNull { search ->
+                search.definedClassName == methodReference.definingClass
+                        && search.methodName == methodReference.name
+                        && methodReference.parameterTypes.toTypedArray().contentEquals(search.methodParams)
+            }
+        }
+    }
+
+    override fun execute(context: BytecodeContext): PatchResult {
+        // Find all instructions where one of the methods is called
+        buildMap {
+            context.classes.forEach { classDef ->
+                if (classDef.type == INTEGRATIONS_CLASS_DESCRIPTOR) {
+                    // avoid infinite recursion
```

Comment:
This code has a lot of similarity with `RemoveScreenshotRestrictionPatch`. For that reason it is advised to make use of OOP patterns and abstract this patch. As reference `AbstractIntegrationsPatch` can be used.

Reply:
is it fine like this?